File: SafeDose-AI/backend/src/SafeDose.Domain/Entities/PatientMedication.cs


namespace SafeDose.Domain.Entities
{
    public class PatientMedication
    {
        public int PatientMedicationId { get; set; }
        public int PatientId { get; set; }
        public int DrugId { get; set; }
        public int? PrescriptionId { get; set; }
        public string? Dose { get; set; }
        public int? Frequency { get; set; }
        public DateOnly? StartDate { get; set; }
        public DateOnly? EndDate { get; set; }
        public byte? MealTiming { get; set; }
        public byte Status { get; set; }

        public Patient Patient { get; set; } = null!;
        public Drug Drug { get; set; }
        public ICollection<PatientMedicationTime> PatientMedicationTimes { get; set; } = [];
        public ICollection<ReminderResponse> ReminderResponses { get; set; } = [];
    }
}

File: SafeDose-AI/frontend/src/app/core/models/patient-medication.ts
export interface PatientMedication {
  patientMedicationId: number;
  patientId: number;
  drugId: number;
  drugName: string;
  drugDose: string;
  prescriptionId: number | null;
  dose: string;
  frequency: number;
  startDate: string;
  endDate: string;
  mealTiming: number;
  status: number;
  statusArabic: string;
  mealTimingArabic: string;
  isVerified: boolean;
  verificationLabelArabic: string;
  drugCatalogId: number;
  times: string[];
}

export interface CreateMedicationPayload {
  patientId: number;
  drugName: string;
  drugCatalogId: number;
  route: number;
  dose: string;
  doctorName: string;
  frequency: number;
  startDate: string;
  endDate: string;
  mealTiming: number;
  prescriptionId: number | null;
  times: string[];
}
